stk/molecular/molecules/molecule.py

```diff
@@ -1050,6 +1051,72 @@ def update_from_mol(self, path, conformer=-1):
         self.mol.RemoveConformer(conformer)
         self.mol.AddConformer(conf)
 
+    def update_from_xyz(self, path, conformer=-1):
+        """
+        Updates molecular structure to match a ``.xyz`` file.
+
+        Parameters
+        ----------
+        path : :class:`str`
+            The full path of the ``.mol`` file from which the structure
+            should be updated.
+
+        conformer : :class:`int`, optional
+            The conformer to be updated.
+
+        Returns
+        -------
+        None : :class:`NoneType`
+
+        Raises
+        ------
+        :class:`RuntimeError`
+            If the number of atoms in the file does not match the
+            number of atoms in the molecule or if atom elements in the
+            file do not agree with the atom elements in the molecule.
+
+        """
+
+        if conformer == -1:
+            conformer = self.mol.GetConformer(conformer).GetId()
+
+        with open(path, 'r') as f:
+            atom_count, _, *content = f.readlines()
+
+        # Check the atom count is correct.
+        num_atoms = self.mol.GetNumAtoms()
+        if int(atom_count) != num_atoms:
+            raise RuntimeError(
+                'The number of atoms in the xyz file does '
+                'not match the number of atoms in the molecule.'
+            )
+
+        # Save all the coords in the file.
+        new_coords = []
+        for i, line in enumerate(content):
+            element, *coords = line.split()
+            if element.isnumeric():
```

Comment:
should raise an error if statement is false

Reply:
This if statement just converts the atomic number into the element symbol, it then compares the element symbol. If this is false, it means that the element symbol was there already so no conversion needs to be done. Of course, if the element symbol is not there but some garbage is, the code necessary to raise an error is already present.